src/dispatch/plugins/dispatch_slack/incident/interactive.py

```diff
@@ -629,35 +668,42 @@ async def handle_participant_role_activity(ack, db_session, context, user):
             for participant_role in active_participant_roles:
                 participant_role.activity += 1
 
-            # re-assign role once threshold is reached
-            if participant_role.role == ParticipantRoleType.observer:
-                if participant_role.activity >= 10:  # ten messages sent to the incident channel
-                    # we change the participant's role to the participant one
-                    participant_role_service.renounce_role(
-                        db_session=db_session, participant_role=participant_role
-                    )
-                    participant_role_service.add_role(
-                        db_session=db_session,
-                        participant_id=participant.id,
-                        participant_role=ParticipantRoleType.participant,
-                    )
-
-                    # we log the event
-                    event_service.log_incident_event(
-                        db_session=db_session,
-                        source="Slack Plugin - Conversation Management",
-                        description=(
-                            f"{participant.individual.name}'s role changed from {participant_role.role} to "
-                            f"{ParticipantRoleType.participant} due to activity in the incident channel"
-                        ),
-                        incident_id=context["subject"].id,
-                    )
+                # re-assign role once threshold is reached
```

Comment:
Looks like we merged the `increment|assess` participant message functions (https://github.com/Netflix/dispatch/blob/a9045d63419b34de4402ffb8a80e1833de32669f/src/dispatch/plugins/dispatch_slack/events.py#L198-L266) here into one. Not sure if that is expected. Resolved unbound participant role here.

Reply:
Yeah, I'll take a look at it.

Reply:
Yes, this was on purpose, as incrementing and role changes are tightly coupled.

Reply:
Added a missing db commit and fixed the docstring